### PR TITLE
Extend ORKTableStepSource for custom section header

### PR DESCRIPTION
**Pull request by tobiasjungnickel**
extend ORKTableStepSource to return custom title and view for header in section

### CODE REVIEW

**Comment by syoung-smallwisdom** [comment]
@tobiasjungnickel This is a great addition. Thanks!

@umerkhan-apple Please take a look.

File: ResearchKit/Common/ORKTableStep.h [hunk 1]
```diff
@@ -84,7 +84,27 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @optional
 - (void)registerCellsForTableView:(UITableView *)tableView;
-    
+
+/**
+ Optional override to return custom header title for section. The default return `nil`.
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKTableStep.h]: "The default returns `nil`"

File: ResearchKit/Common/ORKTableStep.h [hunk 2]
```diff
@@ -84,7 +84,27 @@ NS_ASSUME_NONNULL_BEGIN
...
+ 
+ @param tableView       The table view for custom section header title
+ @param section         The section for this custom header title
+ @return                The custom header title for this section
+ */
+@optional
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section;
+
+/**
+ Optional override to return custom header view for section. The default return `nil`.
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKTableStep.h]: "The default returns nil"

**Comment by syoung-smallwisdom** [comment]
@tobiasjungnickel I discovered a problem with this PR when I merged it into our fork. I've submitted a pull request against your branch to fix this. If you merge that PR, it will automatically show up here.

**Comment by umerkhan-apple** [comment]
@tobiasjungnickel Whenever you have a chance to review @syoung-smallwisdom PR against your branch, can you ping me on this thread? Thanks!

**Comment by tobiasjungnickel** [comment]
@umerkhan-apple merged the PR into my branch now

**Comment by syoung-smallwisdom** [comment]
👍

**Pull request merged**